python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -494,15 +532,44 @@ def _get_threshold(nth_order):
                     # No need to do any outlier rejection assume
                     # PSF to be average of few stars.
                     piffConfig['max_iter'] = 1
+                else:
+                    raise PiffTooFewGoodStarsError(
+                        num_good_stars=len(stars),
+                        minimum_dof=threshold,
+                        poly_ndim=piffConfig['interp']['order'],
+                    )
 
         self._piffConfig = piffConfig
         piffResult = piff.PSF.process(piffConfig)
         wcs = {0: gswcs}
 
         piffResult.fit(stars, wcs, pointing, logger=self.piffLogger)
+
+        nUsedStars = len([s for s in piffResult.stars if not s.is_flagged and not s.is_reserve])
+
+        if piffConfig['interp']['type'] in ['BasisPolynomial', 'Polynomial']:
+            threshold = _get_threshold(piffConfig['interp']['order'])
+            if nUsedStars < threshold:
+                if self.config.zerothOrderInterpNotEnoughStars:
+                    self.log.warning(
+                        "Only %d after outlier rejection, "
+                        "but %d required. Using zeroth order interpolation."%((nUsedStars, threshold))
+                    )
+                    piffConfig['interp']['order'] = 0
+                    # No need to do any outlier rejection assume
+                    # PSF to be average of few stars.
+                    piffConfig['max_iter'] = 1
+                    piffResult.fit(stars, wcs, pointing, logger=self.piffLogger)
+                else:
+                    raise PiffTooFewGoodStarsError(
+                        num_good_stars=nUsedStars,
+                        minimum_dof=threshold,
+                        poly_ndim=piffConfig['interp']['order'],
+                    )
+
         drawSize = 2*np.floor(0.5*stampSize/self.config.samplingSize) + 1
 
-        used_image_pos = [s.image_pos for s in piffResult.stars]
+        used_image_pos = [s.image_pos for s in piffResult.stars if not s.is_flagged and not s.is_reserve]
         if flagKey:
             for candidate in psfCandidateList:
                 source = candidate.getSource()
```

Comment:
I don't know how robust this membership checking below is, when you're comparing `galsim.PositionD` objects with floats. I'd feel much more comfortable if we attached an identifier to each star and compared that. I created [DM-49719](https://rubinobs.atlassian.net/browse/DM-49719) to address this later.

Reply:
Yes agree on that. Happy to work on this after this ticket is merged.